test/Microsoft.AspNetCore.Server.KestrelTests/ConnectionAdapterTests.cs

```diff
@@ -109,6 +108,54 @@ public class ConnectionAdapterTests
             }
         }
 
+        [Fact]
+        public async Task ThrowingOnReadDoesNotCauseUnobservedException()
+        {
+            var unobservedExceptionThrown = false;
+
+            EventHandler<UnobservedTaskExceptionEventArgs> unobservedExceptionHandler = (sender, args) =>
+            {
+                unobservedExceptionThrown = true;
+            };
+
+            try
+            {
+                TaskScheduler.UnobservedTaskException += unobservedExceptionHandler;
```

Comment:
Remove the test and instead just have the handler in the sample. It can be used as a quick sanity check if we need to verify behavior. We could also potentially move it to an end to end test instead (part of the server tests).